File: 713-subarray-product-less-than-k/subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int n = nums.size();
        if(k <= 1) return 0;
        int left = 0, count = 0, product = 1;
        for(int i=0; i <n; i++){
            product *= nums[i];
            while(product >= k){
                product /= nums[left];
                left++;
            }
            count += i - left + 1;
        }
        return count;
    }
};

File: 713-subarray-product-less-than-k/subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k <= 1:
            return 0
        cnt = 0
        sub = 1
        l = 0
        for i in range(len(nums)):
            sub *= nums[i]
            while sub >= k:
                sub //= nums[l]
                l += 1
            cnt += i - l + 1
        return cnt
        